zebra-consensus/src/checkpoint/list.rs

```diff
@@ -201,4 +201,25 @@ impl CheckpointList {
     pub fn iter(&self) -> impl Iterator<Item = (&block::Height, &block::Hash)> {
         self.0.iter()
     }
+
+    /// Returns the checkpoint at `height`, as a zero-based index.
+    /// If `height` is not a checkpoint height, returns the checkpoint immediately before that height.
+    pub fn prev_checkpoint_index(&self, height: block::Height) -> usize {
+        let checkpoint_height = self
+            .max_height_in_range(..=height)
+            .expect("checkpoints must start at the genesis block height 0");
+
+        self.0
+            .keys()
+            .position(|key| key == &checkpoint_height)
+            .expect("checkpoints must start at the genesis block height 0")
```

Comment:
Nitpick/Optional: avoid re-iterating through the checkpoints.

```suggestion
        self.0
            .keys()
            .rposition(|&key| key <= height)
            .expect("checkpoints must start at the genesis block height 0")
```